src/SFA.DAS.AdminService.Web/Validators/Roatp/RoatpGatewayApplicationViewModelValidator.cs

```diff
@@ -0,0 +1,96 @@
+            {
+                validationResponse.Errors.Add(new ValidationErrorDetail("GatewayReviewStatus", NoSelectionErrorMessage));
+            }
+
+            if (validationResponse.Errors.Any()) return await Task.FromResult(validationResponse);
+
+
+            switch (viewModel.GatewayReviewStatus)
+            {
+                case GatewayReviewStatus.AskForClarification:
+                    {
+                        if (string.IsNullOrEmpty(viewModel.OptionAskClarificationText))
+                        {
+                            validationResponse.Errors.Add(new ValidationErrorDetail("OptionAskClarificationText", ErrorEnterClarificationComments));
+                        }
+                        else
+                        {
+                            var wordCount = viewModel.OptionAskClarificationText.Split(new[] { " " }, StringSplitOptions.RemoveEmptyEntries).Length;
+                            if (wordCount > 500)
+                            {
+                                validationResponse.Errors.Add(new ValidationErrorDetail("OptionAskClarificationText", TooManyWords));
+                            }
+                        }
+
+
+                        break;
+                    }
+                case GatewayReviewStatus.Declined:
+                    {
+                        if (string.IsNullOrEmpty(viewModel.OptionDeclinedText))
+                        {
+                            validationResponse.Errors.Add(new ValidationErrorDetail("OptionDeclinedText", ErrorEnterDeclinedComments));
+                        }
+                        else
+                        {
+                            var wordCount = viewModel.OptionDeclinedText.Split(new[] { " " }, StringSplitOptions.RemoveEmptyEntries).Length;
+                            if (wordCount > 500)
+                            {
+                                validationResponse.Errors.Add(new ValidationErrorDetail("OptionDeclinedText", TooManyWords));
+                            }
+                        }
+                        break;
+                    }
+                case GatewayReviewStatus.Approved when !string.IsNullOrEmpty(viewModel.OptionApprovedText):
+                    {
+                        var wordCount = viewModel.OptionApprovedText.Split(new[] { " " }, StringSplitOptions.RemoveEmptyEntries).Length;
+                        if (wordCount > 500)
+                        {
+                            validationResponse.Errors.Add(new ValidationErrorDetail("OptionApprovedText", TooManyWords));
+                        }
+
+                        break;
+                    }
+            }
+
+            return await Task.FromResult(validationResponse);
+        }
+    }
+
+}
+
+public interface IRoatpGatewayApplicationViewModelValidator
+{
+    Task<ValidationResponse> Validate(RoatpGatewayApplicationViewModel viewModel);
+
+}
+
```

Comment:
Please add test coverage for this validator

Reply:
Unit test for validator was added